### PR TITLE
Relax CudaUVM requirement

### LINKED ISSUE #16: Make tests pass in non-UVM mode
- [ ] `LinearBVH::bounds()`
  Seems like a design issue. We (probably) want access to bounds both on host and device.

### PR DESCRIPTION
**Pull request by dalg24**
Fixes #16.

### CODE REVIEW

File: src/ArborX_LinearBVH.hpp [hunk 1]
```diff
@@ -58,6 +58,18 @@ class BoundingVolumeHierarchy
...
 
+    Kokkos::View<bounding_volume_type, DeviceType> getBounds() const
+    {
+      Kokkos::View<bounding_volume_type, DeviceType> bounds("bounds");
+      using ExecutionSpace = typename DeviceType::execution_space;
+      auto bvh = *this;
+      Kokkos::parallel_for(
+          ARBORX_MARK_REGION( "copy bounds" ),
+          Kokkos::RangePolicy<ExecutionSpace>( 0, 1 ),
+          KOKKOS_LAMBDA( int ) { bounds() = bvh.bounds(); } );
```
Comment (Rombur) [inline src/ArborX_LinearBVH.hpp]: I am confused are you just doing a copy here? Why can't you use a `deep_copy`?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -7,7 +7,7 @@ if(KOKKOS_DEVICES MATCHES "OpenMP")
   list(APPEND ARBORX_DEVICE_TYPES Kokkos::OpenMP::device_type)
 endif()
 if(KOKKOS_DEVICES MATCHES "Cuda")
-  list(APPEND ARBORX_DEVICE_TYPES Kokkos::CudaUVMSpace::device_type)
+  list(APPEND ARBORX_DEVICE_TYPES Kokkos::Cuda::device_type)
```
Comment (Rombur) [inline test/CMakeLists.txt]: Is this change temporary or do you want to stop testing for UVM?

File: src/ArborX_DistributedSearchTree.hpp [hunk 3]
```diff
@@ -118,7 +123,10 @@ DistributedSearchTree<DeviceType>::DistributedSearchTree(
     // FIXME when we move to MPI with CUDA-aware support, we will not need to
     // copy from the device to the host
     auto boxes_host = Kokkos::create_mirror_view( boxes );
-    boxes_host( comm_rank ) = _bottom_tree.bounds();
+    auto bounds = _bottom_tree.getBounds();
+    auto bounds_host = Kokkos::create_mirror_view( bounds );
+    Kokkos::deep_copy( bounds_host, bounds );
+    boxes_host( comm_rank ) = bounds_host();
```
[the remark this answers is not in the document]
Comment (dalg24) [reply src/ArborX_DistributedSearchTree.hpp]: Can't remember if we configured MPI to be CUDA aware

File: src/ArborX_LinearBVH.hpp (hunk 1 above)
Comment (dalg24) [reply src/ArborX_LinearBVH.hpp]: I could but technically I only need to feel one entry in the vector.  Will change

File: test/CMakeLists.txt (hunk 2 above)
Comment (dalg24) [reply test/CMakeLists.txt]: We will need to test for both eventually.  Wasn't planning on fixing it in this PR

File: src/ArborX_DistributedSearchTree.hpp [hunk 4]
```diff
@@ -42,7 +42,12 @@ class DistributedSearchTree
     /** Returns the smallest axis-aligned box able to contain all the objects
      *  stored in the tree or an invalid box if the tree is empty.
      */
-    bounding_volume_type bounds() const { return _top_tree.bounds(); }
+    bounding_volume_type bounds() const {
```
[the remark this answers is not in the document]
Comment (dalg24) [reply src/ArborX_DistributedSearchTree.hpp]: I am still thinking about it, I need to come up with something better.

File: src/ArborX_LinearBVH.hpp (hunk 1 above)
Comment (dalg24) [reply src/ArborX_LinearBVH.hpp]: Oh it's not a vector you are 100% right sorry

**Comment by dalg24** [comment]
Opted for a different solution: `BVH::bounds()` is now a host function

File: test/CMakeLists.txt (hunk 2 above)
Comment (Rombur) [reply test/CMakeLists.txt]: We should open an issue about it otherwise we will forget/

File: src/details/ArborX_DetailsTreeConstruction.hpp [hunk 5]
```diff
@@ -442,9 +442,9 @@ class CalculateInternalNodesBoundingVolumesFunctor
   void operator()(int const i) const
   {
     Node *node = _root + _parents(i);
-    // Walk toward the root but do not actually process it because its
+    // Walk toward the root and do process it even though technically its
```
Comment (Rombur) [inline src/details/ArborX_DetailsTreeConstruction.hpp]: Why did you change this?

File: src/ArborX_LinearBVH.hpp [hunk 6]
```diff
@@ -119,6 +112,7 @@ class BoundingVolumeHierarchy
   }
 
   size_t _size;
+  bounding_volume_type _bounds;
```
Comment (aprokop) [inline src/ArborX_LinearBVH.hpp]: Rename to `_scene_bounds`?

File: src/ArborX_LinearBVH.hpp [hunk 7]
```diff
@@ -49,14 +49,7 @@ class BoundingVolumeHierarchy
...
 
-  KOKKOS_INLINE_FUNCTION
-  bounding_volume_type bounds() const
-  {
-    // NOTE should default constructor initialize to an invalid geometry?
-    if (empty())
-      return bounding_volume_type();
-    return getBoundingVolume(getRoot());
-  }
+  bounding_volume_type bounds() const { return _bounds; }
```
Comment (aprokop) [inline src/ArborX_LinearBVH.hpp]: Add a comment about it being a host function.
